### PR TITLE
Fix mkfiles.sh script for Windows

### PR DESCRIPTION
**Pull request by newhoggy**

### CODE REVIEW

File: scripts/byron-to-alonzo/mkfiles.sh [hunk 1]
```diff
@@ -59,12 +59,23 @@ FUNDS_PER_BYRON_ADDRESS=$((${FUNDS_PER_GENESIS_ADDRESS} - 1000000))
...
+UNAME=$(uname -s) DATE=
+case $UNAME in
+  Darwin )      DATE="gdate";;
+  Linux )       DATE="date";;
+  MINGW64_NT* ) UNAME="Windows_NT"
+                DATE="date";;
 esac
 
+sprocket() {
+  if [ "$OS" == "Windows_NT" ]; then
```
Comment (mkoura) [inline scripts/byron-to-alonzo/mkfiles.sh:71]: Shouldn't this use `"$UNAME"`?

**Review by mkoura — APPROVED** [review]
LGTM. I haven't ran it on Windows though.

**Comment by newhoggy** [comment]
bors merge

**Review by Jimbo4350 — APPROVED** [review]

File: scripts/byron-to-alonzo/mkfiles.sh [hunk 2]
```diff
@@ -59,12 +59,23 @@ FUNDS_PER_BYRON_ADDRESS=$((${FUNDS_PER_GENESIS_ADDRESS} - 1000000))
...
+case $UNAME in
+  Darwin )      DATE="gdate";;
+  Linux )       DATE="date";;
+  MINGW64_NT* ) UNAME="Windows_NT"
+                DATE="date";;
 esac
 
+sprocket() {
+  if [ "$UNAME" == "Windows_NT" ]; then
+    echo -n "\\\\.\\pipe\\"
```
Comment (DavidEichmann) [inline scripts/byron-to-alonzo/mkfiles.sh:72]: Can also use single quotes

```bash
echo -n '\\.\pipe\'
```

File: scripts/byron-to-alonzo/mkfiles.sh [hunk 3]
```diff
@@ -476,13 +487,14 @@ mkdir -p run
 
 for NODE in ${BFT_NODES}; do
   (
+    SPROCKET="$(sprocket "${ROOT}/${NODE}/node.sock")"
```
Comment (DavidEichmann) [inline scripts/byron-to-alonzo/mkfiles.sh:490]: Do you need to define a variable? Can you not use this inline. Likewise in the next hunk.

File: scripts/byron-to-alonzo/mkfiles.sh [hunk 4]
```diff
@@ -59,12 +59,23 @@ FUNDS_PER_BYRON_ADDRESS=$((${FUNDS_PER_GENESIS_ADDRESS} - 1000000))
...
-  * )            DATE="date";;
+UNAME=$(uname -s) DATE=
+case $UNAME in
+  Darwin )      DATE="gdate";;
+  Linux )       DATE="date";;
+  MINGW64_NT* ) UNAME="Windows_NT"
+                DATE="date";;
 esac
 
+sprocket() {
```
Comment (DavidEichmann) [inline scripts/byron-to-alonzo/mkfiles.sh:70]: I'd like to see some documentation explaining why we need to do this. Based on your changes and the name of the PR, it looks like we cant just use a file path for the socket? I have no idea what that path `\\.\pipe\` refers to on windows. Is this some special path for pipes (or socket equivalent for windows)? I assume forward slashes really don't work?

**Review by DavidEichmann — CHANGES_REQUESTED** [review]

**Comment by DavidEichmann** [comment]
@newhoggy Have you run this on windows?

File: scripts/byron-to-alonzo/mkfiles.sh (hunk 4 above)
Comment (newhoggy) [reply scripts/byron-to-alonzo/mkfiles.sh:70]: Windows is weird.

So on Windows, we're normally used to paths looking like this: `C:\MyDirectory\MyFile`, but this is only what the user sees.

Filenames actually look like this to the operating system filesystem: `\\?\C:\MyDirectory\MyFile` and `\\.\C:\MyDirectory\MyFile`

This way of writing filenames is called the UNC Path or the device path and allows you to refer to things that are not files.  For example named pipes.

So long story short, Windows does not follow the Unix philosophy where everything is a file.  Instead the devices and non-file resources are referenced via a schema that embeds the filesystem but reserves other namespaces for non-files.

**Comment by newhoggy** [comment]
> Have you run this on windows?

This PR was tested in a Windows VM 😁

File: scripts/byron-to-alonzo/mkfiles.sh (hunk 3 above)
Comment (newhoggy) [reply scripts/byron-to-alonzo/mkfiles.sh]: Inlined!

File: scripts/byron-to-alonzo/mkfiles.sh (hunk 2 above)
Comment (newhoggy) [reply scripts/byron-to-alonzo/mkfiles.sh]: Fixed!

File: scripts/byron-to-alonzo/mkfiles.sh (hunk 4 above)
Comment (newhoggy) [reply scripts/byron-to-alonzo/mkfiles.sh:70]: I've added a short explanation as a comment to the script.

Comment (DavidEichmann) [reply scripts/byron-to-alonzo/mkfiles.sh:70]: Amazing thanks!

**Review by DavidEichmann — APPROVED** [review]

**Comment by newhoggy** [comment]
bors merge
